utils/parsers/npm_parser.py

```diff
@@ -20,45 +23,119 @@ def get_package_info(cls, name, version=None):
+            package_info_version = package_info['versions'][package_version]
+
+            if 'license' in package_info_version:
+                if type(package_info_version['license']) is dict:
+                    info[name][strings.LICENSE] = package_info_version['license']['type']
                 else:
-                    info[strings.LICENSE] = package_info['license']
-            if 'version' in package_info:
-                info[strings.VERSION] = package_info['version']
-            else:
-                if 'dist-tags' in package_info:
-                    if 'latest' in package_info['dist-tags']:
-                        info[strings.VERSION] = package_info['dist-tags']['latest']
-            if 'homepage' in package_info:
-                info[strings.URL] = package_info['homepage']
+                    info[name][strings.LICENSE] = package_info_version['license']
+            if 'homepage' in package_info_version:
+                info[name][strings.URL] = package_info_version['homepage']
+            if depth > 0:
+                for section in npm_sections:
+                    if section in package_info_version:
+                        for dep_name in package_info_version[section].keys():
+                            dep_version = package_info_version[section][dep_name]
+                            dep_info = NpmPackageParser.get_package_info(
+                                dep_name.lower(), dep_version, depth-1, npm_sections, use_semver=use_semver
+                            )
+                            info = {**dep_info, **info}
         else:
-            info[strings.ERROR] = 'Package not found in NPM registry'
-        if version:
-            info[strings.VERSION] = version
+            info[name][strings.ERROR] = 'Package not found in NPM registry'
+
         return info
 
     @classmethod
-    def parse_version(cls, version):
-        regex = r"(?P<spec>[=<>~^]*)(?P<ver>[.a-zA-Z0-9_-]*)"
-        parts = re.match(regex, version)
-        if parts:
-            spec = parts.group('spec').strip()
-            ver = parts.group('ver')
-            version = ver
-            if spec and ver:
-                if spec in ['>=', '>']:
-                    version = None
-
-            return version
-        raise ValueError
+    def satisfies(cls, version, version_spec, use_semver=False):
+        if use_semver:
+            # js2py isn't perfect so we treat special cases to prevent an infinite loop...
+            regex = r"(?P<maj>[0-9]+)\.(?P<min>[0-9]+)\.(?P<pat>[0-9]+)-?(?P<tag>[.a-z0-9]*)"
+            # if only a version is given in the spec, then only this version is valid
+            match = re.match(regex, version_spec)
+            if match is not None and match.span() == (0, len(version_spec)):
+                return version == version_spec
+            if version_spec == '^' + version:
+                return True
+            return semver.satisfies(version, version_spec)
```

Comment:
What happens if semver is not installed? Do you check for that somewhere?

Reply:
I fixed the initialization of `semver` at the beginning of the file (as we don't want to do it if `use_semver` is false). However, it seems that `semver` is automatically installed by `js2py` so there is no need to check this.

Nevertheless, if one tries to run the script with `use_semver=true` without Node installed, one will get this: `AssertionError: You must have node installed! run: brew install node`. I think the error is explicit enough so we don't have to check this?